index.js

```diff
@@ -84,18 +84,88 @@ module.exports = {
+    let appOptions = (this.app && this.app.options) || {};
+    let customOptions = appOptions['ember-cli-babel'];
+
+    if (customOptions && 'includeExternalHelpers' in customOptions) {
+      return customOptions.includeExternalHelpers === true;
+    } else {
+      return false;
+    }
+  },
+
+  _importHelpers(app) {
+    let helpersPath = 'vendor/babel-helpers/babel-helpers.js';
+
+    if (this.import) {  // support for ember-cli >= 2.7
+      this.import(helpersPath, { prepend: true });
+    } else if (app.import) { // support ember-cli < 2.7
+      app.import(helpersPath, { prepend: true });
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn('Please run: ember install ember-cli-import-polyfill');
+    }
+  },
+
+  _getHelperWhitelist() {
+    let appOptions = this._getAppOptions();
+    let customOptions = appOptions['ember-cli-babel'];
+
+    return customOptions && customOptions.externalHelperWhitelist;
+  }
+
+  _getHelperVersion() {
+    let appOptions = this._getAppOptions();
+    let customOptions = appOptions['ember-cli-babel'];
+
+    return customOptions && customOptions.externalHelperVersion;
+  }
+
+  _getHelpersPlugin() {
+    return [
+      [
+        require.resolve('@babel/plugin-external-helpers'),
+        {
+          helperVersion: this._getHelperVersion(),
+          whitelist: this._getHelperWhitelist()
+        }
+      ]
+    ]
+  },
+
   treeForVendor() {
-    if (!this._shouldIncludePolyfill()) { return; }
+    let trees = [];
 
-    const Funnel = require('broccoli-funnel');
-    const UnwatchedDir = require('broccoli-source').UnwatchedDir;
+    if (this._shouldIncludePolyfill()) {
+      const Funnel = require('broccoli-funnel');
+      const UnwatchedDir = require('broccoli-source').UnwatchedDir;
 
-    // Find babel-core's browser polyfill and use its directory as our vendor tree
-    let polyfillDir = path.dirname(require.resolve('@babel/polyfill/dist/polyfill'));
+      // Find babel-core's browser polyfill and use its directory as our vendor tree
+      let polyfillDir = path.dirname(require.resolve('@babel/polyfill/dist/polyfill'));
 
-    return new Funnel(new UnwatchedDir(polyfillDir), {
-      destDir: 'babel-polyfill'
-    });
+      let polyfillTree = new Funnel(new UnwatchedDir(polyfillDir), {
+        destDir: 'babel-polyfill'
+      });
+
+      trees.push(polyfillTree);
+    }
+
+    // Helpers are a global config, so only the root application should bother
+    // generating and including the file.
+    if (this.parent === this.app && this._shouldIncludeHelpers()) {
+      const babel = require('@babel/core');
+      const writeFile = require('broccoli-file-creator');
+
+      let helpers = babel.buildExternalHelpers(this._getHelperWhitelist);
+      let babelHelpersTree = writeFile('babel-helpers/babel-helpers.js', helpers);
+
+      trees.push(babelHelpersTree);
+    }
+
+    if (trees.length > 0) {
+      return new MergeTrees(trees);
+    }
```

Comment:
Don't we need an `else`?

Reply:
the previous early return did `return undefined` so I assumed the return would be optional here

Reply:
ah, gotcha. yeah thats totally fine